examples/multiview.cpp

```diff
@@ -0,0 +1,134 @@
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#include "RAJA/RAJA.hpp"
+#include <cstdio>
+#include <array>
+
+/*
+ * MultiView Usage Example
+ *
+ * A RAJA::MultiView object wraps an array-of-pointers,
+ * or a pointer-to-pointers, whereas a RAJA::View wraps a single
+ * pointer or array. This allows a single RAJA::Layout to be applied to
+ * multiple arrays internal to the MultiView, allowing multiple arrays to share indexing
+ * arithmetic when their access patterns are the same.
+ * 
+ * The instantiation of a MultiView works exactly like a standard View,
+ * except that it takes an array-of-pointers. In the following example, a MultiView
+ * applies a 1-D layout of length 4 to 2 internal arrays in myarr:
+ * 
+ *   int a1[4] = {5,6,7,8}; // Arrays of the same size, which will become internal to the MultiView.
+ *   int a2[4] = {9,10,11,12};
+ *   int * myarr[2]; // Array-of-pointers which will be passed into MultiView.
+ *   myarr[0] = a1;
+ *   myarr[1] = a2;
+ *   // This MultiView applies a 1-D layout of length 4 to each internal array in myarr.
```

Comment:
Adding blank lines in the code example would help readability, similar to the user guide examples. 

Is there a reason you don't include the code snippets in the user guide directly from the example code like we do in other places in the docs?  This makes it easy to keep the example code and docs consistent.

Reply:
The actual example is a bit different; the documentation has has simpler usage, but I can expand on the example and link the documentation to it.

Reply:
@rhornung67 I've addressed the additional cleanup and other issues. Would you like to take one more look?

Reply:
Yes, I will attempt to get to it in between a few more webex meetings this afternoon.

Reply:
Haven't been able to get to it yet. One more webex at 4pm. I will review it tomorrow afternoon. Sorry for the delay. I'm running around like a maniac lately.

Reply:
No worries! This PR is not at the head of the merging queue yet.